heft-plugins/heft-typescript-plugin/src/TypeScriptPlugin.ts

```diff
@@ -203,7 +203,19 @@ export async function loadPartialTsconfigFileAsync(
           },
           jsonPathMetadata: {
             '$.compilerOptions.outDir': {
-              pathResolutionMethod: PathResolutionMethod.resolvePathRelativeToConfigurationFile
+              pathResolutionMethod: PathResolutionMethod.custom,
+              customResolver(
+                resolverOptions: ConfigurationFile.IJsonPathMetadataResolverOptions<IPartialTsconfig>
+              ): string {
+                if (resolverOptions.propertyValue.includes('${configDir}')) {
```

Comment:
Jest configs and API Extractor's config use the `<configDir>` format. We should probably be consistent with what already exists.